### PR TITLE
Explicitly set install_flavor on windows platform

### PR DESCRIPTION
**Pull request by dhoer**
Java can't target multiple platforms due to install_flavor issue #242

### CODE REVIEW

**Comment by dhoer** [comment]
ChefSpec documentation has been tweaked and shows how to keep the resources from an included recipe from being loaded into the Chef run, but test that the recipe was included: 
https://github.com/sethvargo/chefspec/blob/a20f9c47077640292c3445752f6f2f56976f30e3/README.md#include_recipe

File: recipes/default.rb [hunk 1]
```diff
@@ -18,6 +18,8 @@
 # limitations under the License.
 #
 
+node.set['java']['install_flavor'] = 'windows' if platform_family?('windows')
```
Comment (erichelgeson) [inline recipes/default.rb]: I think this is OK, in version 2 we'll refactor this to be less cryptic. Can we add an INFO message here though to let the user know we've changed the node value on them?

**Comment by dhoer** [comment]
Is using log vs. Chef::Log ok? I like log because of chefspec support. 

If you want the log message to change, I can make that change as well.

**Comment by dhoer** [comment]
Closing this because of the addition of mac support.  I handle this situation with the following piece of code:
```
case node[:platform_family]
when 'windows'
  node.set['java']['install_flavor'] = 'windows'
when 'mac_os_x'
  node.set['java']['install_flavor'] = 'homebrew'
end

include_recipe 'java'
```

**Pull request closed without merging**